caliban_toolbox/figure_eight_functions.py

```diff
@@ -29,10 +29,82 @@
 import urllib
+import re
 
 from getpass import getpass
+from urllib.parse import urlencode
+
 from caliban_toolbox.log_file import create_upload_log
 from caliban_toolbox.aws_functions import aws_upload_files, aws_download_files
+from caliban_toolbox.utils.misc_utils import list_npzs_folder
+
+
+def _format_url(aws_folder, stage, npz, url_encoded_dict):
+    base_url = 'https://caliban.deepcell.org/caliban-input__caliban-output__{}__{}__{}?{}'
+    formatted_url = base_url.format(aws_folder, stage, npz, url_encoded_dict)
+
+    return formatted_url
+
+
+def _create_next_log_name(previous_log_name, stage):
+    stage_num = previous_log_name.split('_')[1]
+    new_log = 'stage_{}_{}_upload_log.csv'.format(stage_num + 1, stage)
+
+    return new_log
+
+
+def get_latest_log_file(log_dir):
+    """Find the latest log file in the log directory
+    Args:
+        log_dir: full path to log directory
+    Returns:
+        string: name of the latest log file
+    """
+    files = os.listdir(log_dir)
+    log_files = [file for file in files if 'upload_log.csv' in file]
+    log_files.sort()
+
+    return log_files[-1]
+
+
+def create_job_urls(crop_dir, aws_folder, stage, pixel_only, label_only, rgb_mode):
+    """Helper function to create relevant URLs for caliban log and AWS upload
+    Args:
+        crop_dir: full path to directory with the npz crops
+        aws_folder: path for images to be stored in AWS
+        stage: which stage of the correction process this job is for
+        pixel_only: boolean flag to determine if only pixel mode is available
+        label_only: boolean flag to determine if only label is available
+        rgb_mode: boolean flag to determine if rgb mode will be enabled
+    Returns:
+        list: list of paths to local NPZs to be uploaded
+        list: list of paths to desintation for NPZs
+        list: list of URLs to supply to figure8 to to display crops
+        list: list of NPZs that will be uploaded
+    Raises:
+        ValueError: If URLs are not valid
+    """
+    # TODO: check that URLS don't contain invalid character
+    # load the images from specified folder but not the json log file
+    npzs_to_upload = list_npzs_folder(crop_dir)
+
+    # change slashes separating nested folders to underscores for URL generation
+    subfolders = re.split('/', aws_folder)
+    subfolders = '__'.join(subfolders)
+
+    # create dictionary to hold boolean flags
+    url_dict = {'pixel_only': pixel_only, 'label_only': label_only, 'rgb': rgb_mode}
+    url_encoded_dict = urlencode(url_dict)
+
+    # create path to npz, key to upload npz, and url path for figure8
+    npz_paths, npz_keys, url_paths = [], [], []
+    for npz in npzs_to_upload:
+        npz_paths.append(os.path.join(crop_dir, npz))
+        npz_keys.append(os.path.join(aws_folder, stage, npz))
+        url_paths.append(_format_url(subfolders, stage, npz, url_encoded_dict))
+
+    # TODO: think about better way to structure than than many lists
```

Comment:
This should be handled by a crowdsourcing database similar to (or even identical to) the one used for the data_loader